File: 3280-convert-date-to-binary/3280-convert-date-to-binary.java
class Solution {
    public String convertDateToBinary(String date) {
        String s="";
        int i=0;
        for(int j=0; j<date.length(); j++){
            if(date.charAt(j)=='-'){
                String str=Integer.toBinaryString(Integer.parseInt(date.substring(i,j)));
                i=j+1;
                s+=str+'-';
            }
        }
        s+=Integer.toBinaryString(Integer.parseInt(date.substring(i)));
        return s;
    }
}

File: 3280-convert-date-to-binary/3280-convert-date-to-binary.py
class Solution:
    def convertDateToBinary(self, date: str) -> str:
        l=list(date.split('-'))
        for i in range(len(l)):
            l[i]=bin(int(l[i]))[2:]
        s=""
        for i in range(len(l)):
            if i==len(l)-1:
                s+=l[i]
            else:
                s+=l[i]+"-"
        return s